src/main/scala/chisel3/util/AttributeAnnotation.scala

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.utils
+
+import firrtl.AttributeAnnotation
+import firrtl.annotations.Named
+
+import chisel3.Data
+import chisel3.experimental.{annotate, requireIsAnnotatable, ChiselAnnotation}
+
+object annotateAttribute {
+  def apply[T <: Data](target: T, annoString: String): Unit = {
```

Comment:
```suggestion
  def apply(target: Data, annoString: String): Unit = {
```
I don't think we get any benefit from the generic here and it would prevent us from overloading this method for other types (e.g. `RawModule`). You only really need a generic if it somehow affects other arguments (e.g. due to context bounds) or the return type.

Reply:
Makes sense, I'll update this later today and test locally. I ended up getting sidetracked with a bunch of business side things and haven't had a chance to do much technical work over the last few weeks.

Reply:
Thanks! Was not trying to prod you or anything, was just leaving a comment while I was thinking about it.

Reply:
all good! was going to come back to this soon anyway! looks like im not the only one hitting this too, which def gives me energy to get it moving again, haha.

Reply:
currently CIRCT only supports wire, node, reg, and module. should i just go ahead and add overloaded versions of this for those 4 types?